pkg/inframetadata/internal/hostmap/updater.go

```diff
@@ -0,0 +1,89 @@
+package hostmap
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.uber.org/multierr"
+)
+
+// updater handles updates to a hostmetadata payload,
+// recording changes and (non-fatal) errors.
+type updater struct {
+	// changed records if there have been any changes.
+	changed bool
+	// attrs stores attributes to use for update of the payload.
+	attrs pcommon.Map
+	// err records any non-fatal errors.
+	err error
+}
+
+// newUpdater creates a new Updater.
+func newUpdater(res pcommon.Resource) *updater {
+	return &updater{
+		attrs: res.Attributes(),
+	}
+}
+
+// FromValue sets field from value and records changes.
+func (m *updater) FromValue(field *string, value string) {
```

Comment:
Why is `field` a pointer?

Reply:
Because we want to update it, not just use its value.

Reply:
not using updater anymore